### PR TITLE
[SPARK-42041][SPARK-42013][CONNECT][PYTHON] DataFrameReader should support list of paths

### PR DESCRIPTION
**Pull request by zhengruifeng**
### What changes were proposed in this pull request?
DataFrameReader should support list of paths

### Why are the changes needed?
for parity

### Does this PR introduce _any_ user-facing change?
yes

### How was this patch tested?
added  and enabled UT

### CODE REVIEW

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala [hunk 1]
```diff
@@ -668,10 +669,16 @@ class SparkConnectPlanner(session: SparkSession) {
...
           }
         }
-        reader.load().queryExecution.analyzed
+        if (rel.getDataSource.getPathsCount == 0) {
+          reader.load().queryExecution.analyzed
+        } else if (rel.getDataSource.getPathsCount == 1) {
+          reader.load(rel.getDataSource.getPaths(0)).queryExecution.analyzed
+        } else {
+          reader.load(rel.getDataSource.getPathsList.asScala: _*).queryExecution.analyzed
+        }
```
Comment (HyukjinKwon) [inline connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala:678]: Can we just call:

```scala
reader.load(rel.getDataSource.getPathsList.asScala: _*).queryExecution.analyzed
```

alone without checking the length?

File: python/pyspark/sql/connect/readwriter.py [hunk 2]
```diff
@@ -105,10 +106,17 @@ def load(
...
-        if path is not None:
-            self.option("path", path)
 
-        plan = DataSource(format=self._format, schema=self._schema, options=self._options)
+        if path is None:
+            paths = None
+        elif isinstance(path, str):
+            paths = [path]
+        else:
+            paths = path
```
Comment (HyukjinKwon) [inline python/pyspark/sql/connect/readwriter.py:115]:
```suggestion
        paths = path
        if isinstance(path, str):
            paths = [path]
```

**Review by HyukjinKwon — APPROVED** [review]

File: connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala (hunk 1 above)
Comment (zhengruifeng) [reply connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala:678]: good question. I am not very sure that the logics of single path and multi pathes are the same 

https://github.com/apache/spark/blob/595f14881f92a09b03888dd69f69f9e7edf2a060/sql/core/src/main/scala/org/apache/spark/sql/DataFrameReader.scala#L175-L212

Comment (HyukjinKwon) [reply connector/connect/server/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala:678]: 👌

**Comment by HyukjinKwon** [comment]
Merged to master.

**Pull request closed without merging**